pkg/config/loader.go

```diff
@@ -213,6 +213,9 @@ func (l *Loader) mergeFile(file string) error {
 		}
 	}
 
+	// Add the file to loaded config files list:
+	l.config.AppendFile(file)
```

Comment:
Here we can safely manipulate the private fields of the config, I think we don't need this `AppendFile` method, specially not if it is public.